cpp/tests/copying/gather_struct_tests.cpp

```diff
@@ -33,26 +34,35 @@
 
 #include <rmm/device_buffer.hpp>
 
-#include <nvfunctional>
-
 #include <memory>
 
-using vector_of_columns = std::vector<std::unique_ptr<cudf::column>>;
-using cudf::size_type;
+using namespace cudf;
+using namespace cudf::test;
```

Comment:
> I usually don't prefer using namespace at any top level namespaces.

I agree with you there, personally. The global `using` directive is in response to reviewers on other PRs objecting to the repetition of `using` directives in each test. :/

I can change this back.

Reply:
I have an more complex view and opinion :). I always try to put the `using namespace` statement on top of a source file if 2 conditions:
 * That namespace is going to be used throughout the file (such as `cudf::test::` is used nearly at each line)
 * The source file is a test file or something similar, not a library source file

Otherwise, if a namespace is used rarely in a couple of lines in the file, then I will put the statement inside the local function calling the namespace.

For library source files, I never use `using namespace`.

Reply:
@ttnghia hence the mild suggestion. 
I see that `cudf::test::iterators` is used at many places. 
`cudf::test` is used mostly with `using` type aliases in this file (and not library source).

Reply:
I've changed this in 23f52e5 to remove `using namespace cudf;`, and used pointed `using cudf::test::no_nulls()`, etc.

@karthikeyann, is this more palatable?

Reply:
Looks great. 
Though, I didn't expect to remove `using namespace cudf::test::iterators;`.